src/scripts/build_nst_da.py

```diff
@@ -0,0 +1,305 @@
+            total=total_size_in_bytes,
+        )
+        for data in streamer.iter_content(block_size):
+            pbar.update(len(data))
+            f.write(data)
+
+
+def uncompress_file(filename: str | Path) -> None:
+    """Uncompress a file to a directory.
+
+    Args:
+        filename: The path to the file to uncompress.
+    """
+    filename = str(filename)
+    match get_suffix(filename):
+        case ".tar.gz":
+            logger.info(f"Uncompressing {filename}")
+            with tarfile.open(filename, mode="r:gz") as tar:
+                tar.extractall(path=filename.replace(".tar.gz", ""))
+            Path(filename).unlink()
+        case ".tar.xz":
+            logger.info(f"Uncompressing {filename}")
+            with tarfile.open(filename, mode="r:xz") as tar:
+                tar.extractall(path=filename.replace(".tar.xz", ""))
+            Path(filename).unlink()
+        case ".zip":
+            logger.info(f"Uncompressing {filename}")
+            with zipfile.ZipFile(filename, mode="r") as zip:
+                zip.extractall(path=filename.replace(".zip", ""))
+            Path(filename).unlink()
+        case _:
+            pass
+
+
+def reorganise_files(dataset_dir: str | Path) -> None:
+    """Reorganise the files into `train` and `test` directories.
+
+    Args:
+        dataset_dir: The directory that should contain the dataset.
+    """
+    logger.info("Reorganising files")
+
+    data_dir = Path(dataset_dir)
+    train_dir = data_dir / "train"
+    test_dir = data_dir / "test"
+    train_audio_dir = train_dir / "audio"
+    test_audio_dir = test_dir / "audio"
+    train_audio_dir.mkdir(parents=True, exist_ok=True)
+    test_audio_dir.mkdir(parents=True, exist_ok=True)
+
+    for name in DATA_URLS:
+        if not Path(name).exists():
+            continue
+        match name:
+            case "train_metadata":
+                Path(name).rename("metadata")
+                shutil.move("metadata", train_dir)
+            case "train_audio":
+                data_dir = Path(name) / "dk"
+                for audio_dir in data_dir.iterdir():
+                    if not audio_dir.is_dir():
+                        continue
+                    for audio_file in audio_dir.glob("*.wav"):
+                        Path(audio_file).rename(train_audio_dir / audio_file.name)
+                shutil.rmtree(name)
+            case "test_and_errors":
+                data_dir = Path(name) / "supplement_dk"
+
+                temp_test_audio_dir = data_dir / "testdata" / "audio"
+                for audio_dir in temp_test_audio_dir.iterdir():
+                    if not audio_dir.is_dir():
+                        continue
+                    for audio_file in audio_dir.glob("*.wav"):
+                        Path(audio_file).rename(test_audio_dir / audio_file.name)
+                temp_test_metadata_dir = data_dir / "testdata" / "metadata"
+                shutil.move(temp_test_metadata_dir, test_dir)
+
```

Comment:
Maybe its worth including a comment here, or further up, detailing that this case is handling of error logs from the raw dataset and is not related to any potential errors in the processing of the dataset. I at least was puzzled for a couple second thinking that we would be mixing our error logs with the test-dataset.

Reply:
Included a comment now